core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -193,8 +217,19 @@ private[spark] object HighlyCompressedMapStatus {
     } else {
       0
     }
+    val threshold = SparkEnv.get.conf.get(config.SHUFFLE_ACCURATE_BLOCK_THRESHOLD)
+    val hugeBlockSizesArray = ArrayBuffer[Tuple2[Int, Byte]]()
```

Comment:
If you're only going to be using `avgSize` for the non-huge blocks then I think you'd want to remove the huge blocks from the numerator in that calculation so that you more accurately size the smaller blocks, that way you're left with `averageOfNonEmptyNonHugeBlocks, preciseSizesOfHugeBlocks`.